### PR TITLE
ENH: Extend StainNormalisation Transform to work with Slides Pipeline

### PR DESCRIPTION
**Pull request by kenza-bouzid**
Adjust the input/output shapes.

### CODE REVIEW

File: hi-ml-cpath/src/health_cpath/utils/wsi_utils.py [hunk 1]
```diff
@@ -16,7 +16,10 @@ def image_collate(batch: List) -> Any:
 
     for i, item in enumerate(batch):
         data = item[0]
-        data[SlideKey.IMAGE] = torch.tensor(np.array([ix[SlideKey.IMAGE] for ix in item]))
+        if isinstance(data[SlideKey.IMAGE], torch.Tensor):
+            data[SlideKey.IMAGE] = torch.stack([ix[SlideKey.IMAGE] for ix in item], dim=0)
```
Comment (kenza-bouzid) [inline hi-ml-cpath/src/health_cpath/utils/wsi_utils.py:20]: Stain norm returns tensors

File: hi-ml/src/health_ml/utils/data_augmentations.py [hunk 2]
```diff
@@ -131,14 +131,20 @@ def stain_normalize(img: torch.Tensor, reference_mean: np.ndarray, reference_std
...
         if img.shape[0] > 1:
             for i in range(img.shape[0]):
                 img_tile = img[i]
                 img[i] = self.stain_normalize(img_tile.unsqueeze(0), self.reference_mean, self.reference_std)
             return img
         else:
-            return self.stain_normalize(img, self.reference_mean, self.reference_std)
+            img = self.stain_normalize(img, self.reference_mean, self.reference_std)
+            if len(original_shape) == 3:
+                return img.squeeze(0)
```
Comment (kenza-bouzid) [inline hi-ml/src/health_ml/utils/data_augmentations.py:146]: Collate function above expect images (C, H, W) so we need to unsqueeze the image back

Comment (harshita-s) [reply hi-ml/src/health_ml/utils/data_augmentations.py:146]: unsqueeze or squeeze? shouldn't this be `if len(original_shape) == 4` if there is already an existing batch dimension and you want to squeeze?

Comment (kenza-bouzid) [reply hi-ml/src/health_ml/utils/data_augmentations.py:146]: If there's already a batch dimension, I don't want to remove it, I only squeeze when original_shape==3 to put it back to 3 cause I add it line 136

**Review by harshita-s — APPROVED** [review]

**Pull request merged**